### ISSUE TITLE
unexpected keyword argument in __init__

### ISSUE THREAD

**Issue by milkpirate** [TylerTemp/docpie#18]
Hey,

I am having trouble with `docpie` running in the CI. Please have a look at following job: https://gitlab.com/milkpirate/yeet/-/jobs/251075102

```
______________ test_component_switch_on_off[pwr_state1---host-on] ______________
tests/test_component.py:169: in test_component_switch_on_off
    yeet.main(argv=cmd)
src/yeet.py:71: in main
    version='yeet 1.0',
venv3/lib/python3.7/site-packages/docpie/__init__.py:131: in docpie
    pie = Docpie(**kwargs)
E   TypeError: __init__() got an unexpected keyword argument 'kwargs'
```

Locally everything works as expected, just in the CI it behaves stubborn.

Code to be tested:
```
def main(
    argv: List[str]=sys.argv
):
    args = docpie.docpie(
        __doc__,
        argv=argv,
        help=True,
        version='yeet 1.0',
    )
...
```

Code to test it (e.g.):
```
...
    cmd = 'yeet.py alarm'
    cmd = shlex.split(cmd)
    yeet.main(argv=cmd)
...
```

`docpie==0.4.2`
`python==3.7.3` (local)
`python==3.7.0` (CI)

Could you help me sorting this out?

**Comment by TylerTemp** [comment]
Hi, thanks for reporting, and

## Can not reproduce ##

I can not reproduce the issue whit the following code:

file: yeet.py

```python
"""
Usage: yeet <cmd>
"""

import sys
import docpie

def main(
            argv
        ):
    args = docpie.docpie(
        __doc__,
        argv=argv,
        help=True,
        version='yeet 1.0',
    )
    print(args)
```

file: run.py

```python
import shlex
import yeet

cmd = 'yeet.py alarm'
cmd = shlex.split(cmd)
yeet.main(argv=cmd)
```

run with

```bash
$ python run.py
{'--': False,
 '<cmd>': 'alarm'}
$ python -V                 
Python 3.7.2
```

works fine.

I also cloned the project and tried as follow:

```
git clone git@gitlab.com:milkpirate/yeet.git
python -m venv ./venv3
source venv3/bin/activate
pip install -r ./tests/requirements_test.txt
```

uncomment line 66-69 in file `yeet.py`, and run

```
pytest
```

and all passed.

Please provide more information about how to reproduce this issue.

## About log ##

According to [the log](https://gitlab.com/milkpirate/yeet/-/jobs/251075102),  all says:

```
tests/test_component.py:196: in test_component_switch_toggle_all
    yeet.main(argv=cmd)
src/yeet.py:71: in main
    version='yeet 1.0',
venv3/lib/python3.7/site-packages/docpie/__init__.py:131: in docpie
    pie = Docpie(**kwargs)
E   TypeError: __init__() got an unexpected keyword argument 'kwargs'
```

maybe provide the arguments that passed to `docpie.docpie`, e.g.

```python
print('__doc__=%r, argv=%r' % (__doc__, argv))
args = docpie.docpie(
    __doc__,
    argv=argv,
    help=True,
    version='yeet 1.0',
)
```

**Comment by TylerTemp** [comment]
According to [the failed job](https://gitlab.com/milkpirate/yeet/-/jobs/251075102) that it failed at comment `fe9b0c6a627dd6962312220269396bb13f7f0b17`

```
git checkout fe9b0c6a627dd6962312220269396bb13f7f0b17
pytest
```

works for me here

information:

```
platform linux -- Python 3.7.2, pytest-3.5.0, py-1.8.0, pluggy-0.6.0 -- /tmp/yeet/venv3/bin/python
cachedir: .pytest_cache
rootdir: /tmp/yeet, inifile: pytest.ini
plugins: xdist-1.22.2, forked-1.0.2, cov-2.5.1
```

Please provide a way that I can reproduce this issue.

**Comment by TylerTemp** [comment]
Tried with Python 3.7.0, pytest-3.5.0, py-1.8.0, pluggy-0.6.0 (same as the log says), still no luck


```
$ git checkout fe9b0c6a627dd6962312220269396bb13f7f0b17
HEAD is now at fe9b0c6 wip
$ pytest            
============================== test session starts ===============================
platform linux -- Python 3.7.0, pytest-3.5.0, py-1.8.0, pluggy-0.6.0
rootdir: /tmp/yeet, inifile: pytest.ini
plugins: xdist-1.22.2, forked-1.0.2, cov-2.5.1
collected 69 items                                                               

tests/test_alarm.py ....                                                   [  5%]
tests/test_component.py ..................................                 [ 55%]
tests/test_print_yeelights.py ........                                     [ 66%]
tests/test_switch.py ............                                          [ 84%]
tests/test_turn_off.py .....                                               [ 91%]
src/yeet.py ......                                                         [100%]

------------- generated xml file: /tmp/yeet/reports/yeet_report.xml --------------

----------- coverage: platform linux, python 3.7.0-final-0 -----------
Name          Stmts   Miss  Cover   Missing
-------------------------------------------
src/yeet.py     169      0   100%   
Coverage annotated source written to dir reports/coverage/cov_annotate
Coverage HTML written to dir reports/coverage/cov_html/
Coverage XML written to file reports/coverage/cov.xml

=========================== 69 passed in 1.92 seconds ============================
```

**Comment by milkpirate** [comment]
I cant reproduce it locally, thats what bugs me as well :( Just in the CI its crashing.

I also tried leaving all keywords args from the `docpie` call:
```
def main(
    argv: List[str]=sys.argv
):
    args = docpie.docpie(__doc__)
...
```
Without luck.

Added some prints:
```
def main(
    argv: List[str]=sys.argv
):
    print(__doc__)
    print(argv)

    args = docpie.docpie(
        __doc__,
        argv=argv,
        help=True,
        version='yeet 1.0',
    )
...
```
This is the result:
```
...
©2018 Paul Schroeder - <mail>

['yeet.py', 'switch', 'toggle']
_________________ test_component_switch_toggle_all[pwr_state3] _________________
tests/test_component.py:196: in test_component_switch_toggle_all
    yeet.main(argv=cmd)
src/yeet.py:74: in main
    version='yeet 1.0',
venv3/lib/python3.7/site-packages/docpie/__init__.py:131: in docpie
    pie = Docpie(**kwargs)
E   TypeError: __init__() got an unexpected keyword argument 'kwargs'
----------------------------- Captured stdout call -----------------------------
```
Really strange phenomenon.

**Comment by TylerTemp** [comment]
Can you give me some "get start" document/guidance of gitlab's autotest framework so I can reproduce it there?

**Comment by milkpirate** [comment]
Its not really an autotest feature. It is called CI (continous integration). Basically youre running a docker container in the cloud and set it up to run your tests. Most od the magic happens in the `.gitlab-ci.yaml` file. Every line inder `script` translates to a shell command executed in the docker container.

The container I use for testing is defines here: https://gitlab.com/milkpirate/dockerfiles/blob/master/Dockerfile.python3-pytest-docs it takes an alpine linux base image and installs some systemtools and a bunch of python modules.

I hope this is not too brief...  this seem like a good start: https://www.youtube.com/watch?v=Jav4vbUrqII
If you want to tinker with my repo, you could sign up in gitlab and I give you access to my repo.

BTW gitlab CI is super mighty (and free to some extend - for hobby users totally sufficient), you might really want to learn about. There are alternatives like TravisCI or CircleCI but I like gitlabs the best :)

Just tell me if you need more assistence at any step.

**Comment by TylerTemp** [comment]
Hi, please try docpie version `0.4.3`

Looks like that this issue happens because it uses the function `locals()` inside the `docpie` function. Normally a function's locals will not contain `__init__` variable.

**Comment by milkpirate** [comment]
Thats it! Job successful: https://gitlab.com/milkpirate/yeet/-/jobs/256019079

Thank you very much!!

**Comment by milkpirate** [comment]
Hey Tyler, just FYI, I talked to a collegue who is our python dev. We discussed that bug and he said its like following:
```
129    kwargs = locals()
130    argv = kwargs.pop('argv')
131    pie = Docpie(**kwargs)
```
Line 129: The variable `kwargs` gets initialized (and hence added to the local variables) before you assign the all locals (`=locals()`) as a dict to it. So in `kwargs` is a reference to itself included. Thats most likely causing the `got an unexpected keyword argument 'kwargs'` error. Popping the `kwargs` from `kwargs` just like in line 130 might do the trick to get it working again with the old version.

Thought that might be interesting for you :)

Thanks again.
